lib/ash_phoenix/filter_form/filter_form.ex

```diff
@@ -373,8 +373,12 @@ defmodule AshPhoenix.FilterForm do
     extended_path = path ++ [field]
 
     case Ash.Resource.Info.related(form.resource, extended_path) do
-      nil -> {path, field}
-      _ -> {extended_path, nil}
+      nil ->
+        {path, field}
+
+      related ->
+        %{name: new_field} = hd(Ash.Resource.Info.public_attributes(related)) || nil
```

Comment:
perhaps we should do `hd(Ash.Resource.Info.primary_key(related)`?

Reply:
also the `|| nil` is unnecessary because `hd/1` raises on empty lists.

Reply:
(so maybe do `List.first/1`)

Reply:
> (so maybe do `List.first/1`)

Good catch.

> perhaps we should do `hd(Ash.Resource.Info.primary_key(related)`?

I'm open to that, the reason I didn't was thinking it might leave open the ability to hint which got selected by defining that attribute first. I suppose that's kind of an anti-pattern. What do you think?

Reply:
Yeah, I think we would solve that with config on the filter form somehow, or with some other heuristic.

Reply:
well, I think its safe enough to just use the first attribute, so lets go with that.